<launch>
  <arg name="param_file" default="$(find-pkg-share autoware_ekf_localizer)/config/ekf_localizer.param.yaml"/>

  <arg name="input_initial_pose_name" default="initialpose3d"/>
  <arg name="input_trigger_node_service_name" default="trigger_node" description="trigger node service"/>

  <!-- input topic name -->
  <arg name="input_pose_with_cov_name" default="in_pose_with_covariance"/>
  <arg name="input_twist_with_cov_name" default="in_twist_with_covariance"/>

  <!-- output topic name -->
  <arg name="output_odom_name" default="ekf_odom"/>
  <arg name="output_pose_name" default="ekf_pose"/>
  <arg name="output_pose_with_covariance_name" default="ekf_pose_with_covariance"/>
  <arg name="output_biased_pose_name" default="ekf_biased_pose"/>
  <arg name="output_biased_pose_with_covariance_name" default="ekf_biased_pose_with_covariance"/>
  <arg name="output_twist_name" default="ekf_twist"/>
  <arg name="output_twist_with_covariance_name" default="ekf_twist_with_covariance"/>
  <arg name="output_processing_time_ms" default="debug/processing_time_ms"/>

  <node pkg="autoware_ekf_localizer" exec="autoware_ekf_localizer_node" output="both">
    <remap from="in_pose_with_covariance" to="$(var input_pose_with_cov_name)"/>

    <remap from="in_twist_with_covariance" to="$(var input_twist_with_cov_name)"/>

    <remap from="initialpose" to="$(var input_initial_pose_name)"/>
    <remap from="trigger_node_srv" to="$(var input_trigger_node_service_name)"/>

    <remap from="ekf_odom" to="$(var output_odom_name)"/>
    <remap from="ekf_pose" to="$(var output_pose_name)"/>
    <remap from="ekf_pose_with_covariance" to="$(var output_pose_with_covariance_name)"/>
    <remap from="ekf_biased_pose" to="$(var output_biased_pose_name)"/>
    <remap from="ekf_biased_pose_with_covariance" to="$(var output_biased_pose_with_covariance_name)"/>
    <remap from="ekf_twist" to="$(var output_twist_name)"/>
    <remap from="ekf_twist_with_covariance" to="$(var output_twist_with_covariance_name)"/>
    <remap from="debug/processing_time_ms" to="$(var output_processing_time_ms)"/>

    <param from="$(var param_file)"/>
  </node>
</launch>
